libirods_rule_engine_plugin-elasticsearch.cpp

```diff
@@ -136,12 +141,16 @@ namespace
 			using tcp = net::ip::tcp;
 
 			// This lambda encapsulates the HTTP logic that's independent of the type of stream.
-			const auto construct_and_send_http_request = [_verb, _target, _body](auto& _stream) {
+			const auto construct_and_send_http_request = [_verb, _target, _body, use_tls](auto& _stream) {
 				http::request<http::string_body> req{_verb, _target, 11};
 				req.set(http::field::host, boost::asio::ip::host_name());
 				req.set(http::field::user_agent, "iRODS Indexing Plugin/" IRODS_PLUGIN_VERSION);
 				req.set(http::field::content_type, "application/json");
 
+				if (use_tls && !config->base64_encoded_creds.empty()) {
+					req.set(http::field::authorization, fmt::format("Basic {}", config->base64_encoded_creds));
+				}
```

Comment:
Consider logging a message when the plugin is configured for TLS, but the creds string is empty.

Reply:
Then again, use of TLS does not imply authentication is necessary. That decision is up to the administrator of the ES server.

Authentication is only recommended.

Reply:
No logging necessary.

Done.